pkg/paralleltest/paralleltest.go

```diff
@@ -16,28 +15,36 @@ It also checks that the t.Parallel is used if multiple tests cases are run as pa
 As part of ensuring parallel tests works as expected it checks for reinitialising of the range value
 over the test cases.(https://tinyurl.com/y6555cy6)`
 
-var Analyzer = &analysis.Analyzer{
-	Name:     "paralleltest",
-	Doc:      Doc,
-	Run:      run,
-	Flags:    flags(),
-	Requires: []*analysis.Analyzer{inspect.Analyzer},
+func NewAnalyzer() *analysis.Analyzer {
+	return newParallelAnalyzer().analyzer
 }
 
-const ignoreMissingFlag = "i"
-
-func flags() flag.FlagSet {
-	options := flag.NewFlagSet("", flag.ExitOnError)
-	options.Bool(ignoreMissingFlag, false, "ignore missing calls to t.Parallel")
-	return *options
+// parallelAnalyzer is an internal analyzer that makes options available to a
+// run pass. It wraps an `analysis.Analyzer` that should be returned for
+// linters.
+type parallelAnalyzer struct {
+	analyzer              *analysis.Analyzer
+	ignoreMissing         bool
+	ignoreMissingSubtests bool
```

Comment:
I put in a wrapper struct like this to make it much simpler and less hacky to set extra options from the test suite. The convention is inspired by Forbidigo (another Go linter) whose code I referenced.